viewing_party/party.py

```diff
@@ -1,23 +1,176 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+
+    if not (title and genre and rating):
+        return None
+    else:
+        movie = { "title" : title,
+                "genre" : genre,
+                "rating" : rating
+                }
+        return movie
+    
+def add_to_watched(user_data, movie):
+
+    user_data["watched"].append(movie)
+    return user_data
+
+
+def add_to_watchlist(user_data, movie):
+
+    user_data["watchlist"].append(movie)
+    return user_data
+
+
+def watch_movie(user_data, title):
+
+    for movie in user_data["watchlist"]:
+        if movie["title"] == title:
+            user_data["watchlist"].remove(movie)
+            user_data["watched"].append(movie)
+            break
+
+    return user_data    
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(user_data):
+    rating_list = []
+    if not user_data["watched"]:
+        return 0.0
+    for movie in user_data["watched"]:
+        rating_list.append(movie["rating"])
+    average_rating = sum(rating_list) / len(rating_list)
+    return average_rating

```

Comment:
I think there's always a difference of opinion about when we should pull values out into variables because it does affect readability. However, I do think sum(rating_list) / len(rating_listis simple enough code that you could also write this as:
return sum(rating_list) / len(rating_list)

without creating average_rating, but I don't think it hurts to use the variable either. Something to consider.